core/Period/Range.php

```diff
@@ -244,7 +244,7 @@ protected function generate()
                 }
             }
 
-            $lastN = $this->getMaxN($lastN);
+            $lastN = (int) $this->getMaxN($lastN);
```

Comment:
Would it be better if `getMaxN` had a strict return type of `int` and the conversion happened there?

Reply:
In that case we would need to ensure that the parameter provided to that method already is an integer. But in terms of more stricter type hinting I guess it's worth to change that.